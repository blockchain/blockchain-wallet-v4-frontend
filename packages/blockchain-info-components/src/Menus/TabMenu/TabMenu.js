import React from 'react'
import styled from 'styled-components'

const BaseTabMenu = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
`

const TabMenu = props => {
  const { children } = props

  return <BaseTabMenu>{children}</BaseTabMenu>
}

export default TabMenu
