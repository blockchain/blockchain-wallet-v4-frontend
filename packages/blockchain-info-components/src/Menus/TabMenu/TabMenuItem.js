import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const BaseTabMenuItem = styled.span`
  position: relative;
  padding: 6px 8px;
  font-family: 'Montserrat', sans-serif;
  font-weight: 500;
  font-size: 14px;
  text-align: center;
  color: ${props =>
    props.selected ? props.theme['gray-5'] : props.theme['gray-3']};
  cursor: pointer;

  &:not(:last-child) {
    margin-right: 15px;
  }

  &:hover {
    color: ${props => props.theme['gray-5']};
  }

  &.active {
    color: ${props => props.theme['gray-5']};
    & :after {
      position: absolute;
      content: '';
      top: 40px;
      left: 0;
      width: 100%;
      border-bottom: 4px solid ${props => props.theme['brand-secondary']};
    }
  }
  @media (max-width: 480px) {
    font-size: 13px;
  }
`

const TabMenuItem = props => {
  const { children, selected, ...rest } = props

  return (
    <BaseTabMenuItem selected={selected} {...rest}>
      {children}
    </BaseTabMenuItem>
  )
}

TabMenuItem.propTypes = {
  selected: PropTypes.bool
}

export default TabMenuItem
