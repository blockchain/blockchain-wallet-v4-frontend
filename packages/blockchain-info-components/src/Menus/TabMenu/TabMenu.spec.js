import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TabMenu from './TabMenu'

describe('TabMenu component', () => {
  it('default renders correctly', () => {
    const component = shallow(
      <TabMenu>
        <span>tab</span>
      </TabMenu>
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
