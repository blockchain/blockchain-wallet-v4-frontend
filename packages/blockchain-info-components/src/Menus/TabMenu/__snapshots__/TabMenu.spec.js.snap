// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu component default renders correctly 1`] = `
<TabMenu__BaseTabMenu>
  <span>
    tab
  </span>
</TabMenu__BaseTabMenu>
`;
